main.py

```diff
@@ -358,6 +358,14 @@ async def achemb(message, ach_id, send_type, author_string=None):
 async def cat_type_autocomplete(interaction: discord.Interaction, current: str) -> list[discord.app_commands.Choice[str]]:
     return [discord.app_commands.Choice(name=choice, value=choice) for choice in cattypes if current.lower() in choice.lower()][:25]
 
+# converts string to lowercase alphanumeric characters only
+def alnum(string):
+    return "".join(item for item in string.lower() if item.isalnum())
+
+# function to autocomplete achievement choice for /giveachievement, which also allows more than 25 options
+async def ach_autocomplete(interaction: discord.Interaction, current: str) -> list[discord.app_commands.Choice[str]]:
+    return [discord.app_commands.Choice(name=val, value=val) for (key, val) in ach_list.items() if (alnum(current) in alnum(key) or alnum(current) in alphanumeric(val))][:25]
```

Comment:
val here is the entire `{'title': 'First cat!', 'description': 'The journey begins...', 'is_hidden': False, 'category': 'Cat Hunt'}` dict